### PR TITLE
feat(gh actions): adds an action to aggregate the data PE-1402

### PR DESCRIPTION
**Pull request by matibat**

### CODE REVIEW

File: .github/workflow/aggregate.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+          cache: "yarn"
+
+      - name: Setup dependencies
+        run: yarn install --immutable --check-cache
+        run: yarn add ardrive-cli
+
+      - name: Write wallet
+        run: echo -n '${{ secrets.WALLET_FILE }}' | base64 --decode > /tmp/wallet
+
+      - name: Lint
```
Comment (karlprieb) [inline .github/workflow/aggregate.yaml]: ?? you forgot to remove?

File: .github/workflow/aggregate.yaml [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+        run: yarn lint
+
+      - name: Build
+        run: yarn build
+
+
+      - name: Fetch previous output
+        run: ardrive download-file -f a7afa47b-a16c-44ad-8d87-c63f935af568
```
Comment (karlprieb) [inline .github/workflow/aggregate.yaml]: from where you got this file-id? I made an upload before of the latest json that I generated locally and the file-id is 7fa5d4e3-0087-422a-acb3-2e481d98d08b

File: .github/workflow/aggregate.yaml [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+      - name: Run
+        run: yarn node aggregate
+
+      - name: Publish output
+        run: ardrive upload-file --local-path ./daily_output.json -F "4b4ffe84-a6de-48d0-9b85-c56db14084a5" -w /tmp/wallet --no-bundle --upsert --dry-run
```
Comment (karlprieb) [inline .github/workflow/aggregate.yaml]: folder id should be d62a47c3-0b9d-4442-ac72-252a239d0469

File: .github/workflow/aggregate.yaml [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+      - name: Set Up NodeJS
+        uses: actions/setup-node@v3
+        with:
+          node-version: ${{ steps.node_version.outputs.NODE_VERSION }}
+          cache: "yarn"
+
+      - name: Setup dependencies
+        run: yarn install --immutable --check-cache
+        run: yarn add ardrive-cli
```
Comment (fedellen) [inline .github/workflow/aggregate.yaml:26]: Pretty surprising and also really cool to see this right in the action

File: .github/workflow/aggregate.yaml (hunk 3 above)
Comment (matibat) [reply .github/workflow/aggregate.yaml]:
```suggestion
        run: ardrive upload-file --local-path ./daily_output.json -F "d62a47c3-0b9d-4442-ac72-252a239d0469" -w /tmp/wallet --no-bundle --upsert --dry-run
```

File: .github/workflow/aggregate.yaml (hunk 2 above)
Comment (matibat) [reply .github/workflow/aggregate.yaml]:
```suggestion
        run: ardrive download-file -f 7fa5d4e3-0087-422a-acb3-2e481d98d08b
```

File: .github/workflow/aggregate.yaml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+      - name: Write wallet
+        run: echo -n '${{ secrets.WALLET_FILE }}' | base64 --decode > /tmp/wallet
+
+      - name: Lint
+        run: yarn lint
+
```
Comment (matibat) [inline .github/workflow/aggregate.yaml]:
```suggestion
```

File: src/common.ts [hunk 6]
```diff
@@ -25,3 +25,12 @@ export function getMinBlockHeigh(): number {
...
 	return file.blockHeight + 1;
 }
+
+export function getLastTimestamp(): number {
+	const hasOutputFile = fs.existsSync(OUTPUT_NAME);
+
+	const fileToCheck = hasOutputFile ? OUTPUT_NAME : OUTPUT_TEMPLATE_NAME;
+
+	const file = JSON.parse(fs.readFileSync(fileToCheck).toString());
+	return file.timestamp + 1;
```
Comment (fedellen) [inline src/common.ts]: Why is this `+ 1`?

Comment (fedellen) [reply src/common.ts]: The function above this is missing the `t` on `getMinBlockHeigh`

File: src/common.ts [hunk 7]
```diff
@@ -25,3 +25,12 @@ export function getMinBlockHeigh(): number {
 	const file = JSON.parse(fs.readFileSync(fileToCheck).toString());
 	return file.blockHeight + 1;
 }
+
+export function getLastTimestamp(): number {
```
Comment (fedellen) [inline src/common.ts:29]: When I run the first test after install with no `daily_output.json` file, the tests all pass. But after the first test, this `daily_output.json` file is produced:

```
!{{ NOT A VALID JSON "": false
```

Then this tests fail to complete on this function with this stack trace:

```shell
  1) Uncaught error outside test suite:
     Uncaught SyntaxError: Unexpected token ! in JSON at position 0
      at JSON.parse (<anonymous>)
      at getLastTimestamp (src/common.ts:2:4051)
      at new DailyOutput (src/daily_output.ts:4:346)
      at Suite.<anonymous> (src/daily_output.test.ts:75:17)
      at Object.<anonymous> (src/daily_output.test.ts:74:1)
      at Module._compile (internal/modules/cjs/loader.js:1068:30)
      at Module.replacementCompile (.yarn/cache/append-transform-npm-2.0.0-99bd7d69ed-f26f393bf7.zip/node_modules/append-transform/index.js:60:13)
      at Module.m._compile (.yarn/__virtual__/ts-node-virtual-9002d3d23b/0/cache/ts-node-npm-10.7.0-ef39b1d45e-2a379e43f7.zip/node_modules/ts-node/src/index.ts:1455:23)
```

File: src/common.ts (hunk 6 above)
Comment (matibat) [reply src/common.ts]: 40e258358fc3351a4f007df2ca5cd16926ce2f0f and 65de9aff8f7e6b114c30b4bca1a97e52eaad1229

File: src/common.ts (hunk 7 above)
Comment (matibat) [reply src/common.ts:29]: Remove the file after the tests run: 7dc58d42a626c9e85010858e7552a938533d7ac7

**Review by karlprieb — APPROVED** [review]

File: .github/workflow/aggregate.yaml [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+
+      - name: Fetch previous output
+        run: ardrive download-file -f 7fa5d4e3-0087-422a-acb3-2e481d98d08b
+
+      - name: Run
+        run: yarn node aggregate
+
+      - name: Publish output
+        run: ardrive upload-file --local-path ./daily_output.json -F "d62a47c3-0b9d-4442-ac72-252a239d0469" -w /tmp/wallet --no-bundle --upsert --dry-run
```
Comment (fedellen) [inline .github/workflow/aggregate.yaml:42]: Are we merging this with `--dry-run`?

Comment (matibat) [reply .github/workflow/aggregate.yaml:42]: I want to test it with what's at #12, which changes the action as well

**Review by fedellen — APPROVED** [review]

**Pull request merged**